vtr_flow/scripts/run_vtr_task.py

```diff
@@ -0,0 +1,479 @@
+            num_failed = run_parallel(args, jobs, run_dirs)
+            print(
+                "Elapsed time: {}".format(format_elapsed_time(datetime.now() - start))
+            )
+
+        if args.parse:
+            print("\nParsing test results...")
+            print("scripts/parse_vtr_task.py -l {}".format(args.list_file[0]))
+            parse_tasks(configs, jobs)
+
+        if args.create_golden:
+            create_golden_results_for_tasks(configs)
+
+        if args.check_golden:
+            num_failed += check_golden_results_for_tasks(configs)
+
+        if args.calc_geomean:
+            summarize_qor(configs)
+            calc_geomean(args, configs)
+    elif args.system == "scripts":
+        for directory in run_dirs:
+            Path(directory).mkdir(parents=True)
+        run_scripts = create_run_scripts(args, jobs, run_dirs)
+        for script in run_scripts:
+            print(script)
+    else:
+        raise VtrError("Unrecognized run system {system}".format(system=args.system))
+    return num_failed
+
+
+def run_parallel(args, queued_jobs, run_dirs):
+    """
+    Run each external command in commands with at most args.j commands running in parllel
+    """
+    # Determine the run dir for each config
+
+    # We pop off the jobs of queued_jobs, which python does from the end,
+    # so reverse the list now so we get the expected order. This also ensures
+    # we are working with a copy of the jobs
+    queued_jobs = list(reversed(queued_jobs))
+    # Find the max taskname length for pretty printing
+
+    queued_procs = []
+    queue = Manager().Queue()
+    for job in queued_jobs:
+        queued_procs += [(queue, run_dirs, job, args.script)]
+    # Queue of currently running subprocesses
+
+    num_failed = 0
+    with Pool(processes=args.j) as pool:
+        pool.starmap(run_vtr_flow_process, queued_procs)
+        pool.close()
+        pool.join()
+    for _ in queued_procs:
+        num_failed += queue.get()
+
+    return num_failed
+
+
+def create_run_scripts(args, jobs, run_dirs):
+    """ Create the bash script files for each job run """
+    run_script_files = []
+    for job in jobs:
+        run_script_files += [create_run_script(args, job, job.work_dir(run_dirs[job.task_name()]))]
+    return run_script_files
+
+
+def create_run_script(args, job, work_dir):
+    """ Create the bash run script for a particular job """
+
+    runtime_estimate = ret_expected_runtime(job, work_dir)
+    memory_estimate = ret_expected_memory(job, work_dir)
+    if runtime_estimate < 0:
+        runtime_estimate = 0
+
+    if memory_estimate < 0:
+        memory_estimate = 0
+
+    human_readable_runtime_est = format_human_readable_time(runtime_estimate)
+    human_readable_memory_est = format_human_readable_memory(memory_estimate)
+    Path(work_dir).mkdir(parents=True)
+    run_script_file = Path(work_dir) / "vtr_flow.sh"
+    with open(run_script_file, "w+") as out_file:
+        print("#!/usr/bin/env bash", file=out_file)
```

Comment:
Instead of embedding this, could this be a template that is loaded?
At the very least, could this be joined into one string and one call to `format`?

Reply:
I created a template in the newest commit.